### ISSUE TITLE
error when download any video  from vimeo.com

### ISSUE THREAD

**Issue by DogLi** [soimort/you-get#1998]
```shell
$ you-get https://vimeo.com/215101646 --debug
```
the result is:
```
you-get: version 0.4.750, a tiny downloader that scrapes the web.
you-get: ['https://vimeo.com/215101646']
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/you_get/extractors/vimeo.py", line 34, in vimeo_download_by_id
    config_url = unescape_html(r1(r'data-config-url="([^"]+)"', html))
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/html/__init__.py", line 130, in unescape
    if '&' not in s:
TypeError: argument of type 'NoneType' is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/you-get", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/you_get/__main__.py", line 92, in main
    main(**kwargs)
  File "/usr/local/lib/python3.6/site-packages/you_get/common.py", line 1444, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/you_get/common.py", line 1355, in script_main
    download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "/usr/local/lib/python3.6/site-packages/you_get/common.py", line 1157, in download_main
    download(url, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/you_get/common.py", line 1437, in any_download
    m.download(url, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/you_get/extractors/vimeo.py", line 67, in vimeo_download
    vimeo_download_by_id(id, None, output_dir=output_dir, merge=merge, info_only=info_only, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/you_get/extractors/vimeo.py", line 43, in vimeo_download_by_id
    video_page = get_content('http://player.vimeo.com/video/%s' % id, headers=fake_headers)
  File "/usr/local/lib/python3.6/site-packages/you_get/common.py", line 330, in get_content
    response = urlopen_with_retry(req)
  File "/usr/local/lib/python3.6/site-packages/you_get/common.py", line 307, in urlopen_with_retry
    return request.urlopen(*args, **kwargs)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 526, in open
    response = self._open(req, data)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 544, in _open
    '_open', req)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 1346, in http_open
    return self.do_open(http.client.HTTPConnection, req)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 1321, in do_open
    r = h.getresponse()
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1331, in getresponse
    response.begin()
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 297, in begin
    version, status, reason = self._read_status()
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 258, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/socket.py", line 586, in readinto
    return self._sock.recv_into(b)
ConnectionResetError: [Errno 54] Connection reset by peer
```

**Comment by DogLi** [comment]
yes, it works under the shadowsocks!

**Comment by rslonik** [comment]
Kinda same problem here. Tried from 2 different networks. Note the `TypeError` and not `ConnectionResetError` as DogLi had.

```
root@vs:~# you-get "https://vimeo.com/152225664" --debug
[DEBUG] get_content: https://vimeo.com/152225664
[DEBUG] get_content: http://player.vimeo.com/video/152225664
you-get: version 0.4.750, a tiny downloader that scrapes the web.
you-get: ['https://vimeo.com/152225664']
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/you_get/extractors/vimeo.py", line 34, in vimeo_download_by_id
    config_url = unescape_html(r1(r'data-config-url="([^"]+)"', html))
  File "/usr/lib/python3.5/html/__init__.py", line 130, in unescape
    if '&' not in s:
TypeError: argument of type 'NoneType' is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/you-get", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/dist-packages/you_get/__main__.py", line 92, in main
    main(**kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1444, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1355, in script_main
    download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1157, in download_main
    download(url, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1437, in any_download
    m.download(url, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/extractors/vimeo.py", line 67, in vimeo_download
    vimeo_download_by_id(id, None, output_dir=output_dir, merge=merge, info_only=info_only, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/extractors/vimeo.py", line 45, in vimeo_download_by_id
    info = loads(match1(video_page, r'var t=(\{[^;]+\});'))
  File "/usr/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'NoneType'
```

**Comment by missqso** [comment]
same problem..
tried from two different network that unicom 4g and company wifi.

python version : 3.6.1
you-get version : 0.4.750
os : win10

````
C:\Users\nt>you-get -x 127.0.0.1:4411  https://vimeo.com/198813859  --debug
[DEBUG] get_content: https://vimeo.com/198813859
[DEBUG] get_content: http://player.vimeo.com/video/198813859
you-get: version 0.4.750, a tiny downloader that scrapes the web.
you-get: ['https://vimeo.com/198813859']
Traceback (most recent call last):
  File "c:\users\nt\appdata\local\programs\python\python36-32\lib\site-packages\you_get\extractors\vimeo.py", line 34, in vimeo_download_by_id
    config_url = unescape_html(r1(r'data-config-url="([^"]+)"', html))
  File "c:\users\nt\appdata\local\programs\python\python36-32\lib\html\__init__.py", line 130, in unescape
    if '&' not in s:
TypeError: argument of type 'NoneType' is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\nt\appdata\local\programs\python\python36-32\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\nt\appdata\local\programs\python\python36-32\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\nt\AppData\Local\Programs\Python\Python36-32\Scripts\you-get.exe\__main__.py", line 9, in <module>
  File "c:\users\nt\appdata\local\programs\python\python36-32\lib\site-packages\you_get\__main__.py", line 92, in main
    main(**kwargs)
  File "c:\users\nt\appdata\local\programs\python\python36-32\lib\site-packages\you_get\common.py", line 1444, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "c:\users\nt\appdata\local\programs\python\python36-32\lib\site-packages\you_get\common.py", line 1355, in script_main
    download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "c:\users\nt\appdata\local\programs\python\python36-32\lib\site-packages\you_get\common.py", line 1157, in download_main
    download(url, **kwargs)
  File "c:\users\nt\appdata\local\programs\python\python36-32\lib\site-packages\you_get\common.py", line 1437, in any_download
    m.download(url, **kwargs)
  File "c:\users\nt\appdata\local\programs\python\python36-32\lib\site-packages\you_get\extractors\vimeo.py", line 67, in vimeo_download
    vimeo_download_by_id(id, None, output_dir=output_dir, merge=merge, info_only=info_only, **kwargs)
  File "c:\users\nt\appdata\local\programs\python\python36-32\lib\site-packages\you_get\extractors\vimeo.py", line 45, in vimeo_download_by_id
    info = loads(match1(video_page, r'var t=(\{[^;]+\});'))
  File "c:\users\nt\appdata\local\programs\python\python36-32\lib\json\__init__.py", line 348, in loads
    'not {!r}'.format(s.__class__.__name__))
TypeError: the JSON object must be str, bytes or bytearray, not 'NoneType'
````

**Comment by demonluo** [comment]
can't download any video from vimeo using you-get...
```
https://vimeo.com/79285546
https://vimeo.com/channels/staffpicks/143603739
https://vimeo.com/ondemand/backintime
```

**Comment by rosynirvana** [comment]
```
"embed_code": "&lt;iframe src=&quot;https://player.vimeo.com/video/79285546&quot; width=&quot;640&quot; height=&quot;360&quot; frameborder=&quot;0&quot; webkitallowfullscreen mozallowfullscreen allowfullscreen&gt;&lt;/iframe&gt;",
```

``` info = loads(match1(video_page, r'var t=(\{[^;]+\});'))```

**Comment by rosynirvana** [comment]
```
$ ./you-get -di 'https://vimeo.com/channels/staffpicks/143603739'
[DEBUG] get_content: https://api.vimeo.com/channels/staffpicks/videos?access_token=f6785418277b72c7c87d3132c79eec24
[DEBUG] get_content: https://vimeo.com/224326254
[DEBUG] get_content: http://player.vimeo.com/video/224326254
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4865/8/224326254/785943464.mp4?token=1499607361-0x905e1559149482286bbc5f8a5d9ad749df3c53b1
Site:       Vimeo.com
Title:      YES, GOD, YES from Karen Maine on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       364.43 MiB (382128716 Bytes)

[DEBUG] get_content: https://vimeo.com/224136480
[DEBUG] get_content: http://player.vimeo.com/video/224136480
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4827/8/224136480/785088662.mp4?token=1499607368-0xc9bc0912e8b30791a16345c435041e3d6eaa1d4b
Site:       Vimeo.com
Title:      △▽ от Conner Griffith on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       30.04 MiB (31502296 Bytes)

[DEBUG] get_content: https://vimeo.com/224011571
[DEBUG] get_content: http://player.vimeo.com/video/224011571
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4802/8/224011571/784505155.mp4?token=1499607375-0x202a32af54aa9365d1b40caea02956c86ac4b5e2
Site:       Vimeo.com
Title:      Hot Dog Hands from Matt Reynolds on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       249.92 MiB (262060402 Bytes)

[DEBUG] get_content: https://vimeo.com/223912295
[DEBUG] get_content: http://player.vimeo.com/video/223912295
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4782/8/223912295/787094107.mp4?token=1499607386-0x9740dacd39a0a196da81b46257d14312e15ef258
Site:       Vimeo.com
Title:      Culture Shifters: Jake Burghart (Director of Photography @ Vice) from Mike Pagan on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       263.24 MiB (276029317 Bytes)

[DEBUG] get_content: https://vimeo.com/223810249
[DEBUG] get_content: http://player.vimeo.com/video/223810249
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4762/8/223810249/783697660.mp4?token=1499607394-0x09e4b3762cbfb8551fc40b086ce42c810217c482
Site:       Vimeo.com
Title:      Washed Out: Floating By from Drew Tyndell on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       146.14 MiB (153242342 Bytes)

[DEBUG] get_content: https://vimeo.com/223768305
[DEBUG] get_content: http://player.vimeo.com/video/223768305
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4753/8/223768305/783404277.mp4?token=1499607403-0xf724618e390725160f2613d652940a6288909ac8
Site:       Vimeo.com
Title:      Supernaive - Opal Waltz (2017) de Vincent Tsui on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       24.77 MiB (25976262 Bytes)

[DEBUG] get_content: https://vimeo.com/223715406
[DEBUG] get_content: http://player.vimeo.com/video/223715406
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4743/8/223715406/783151839.mp4?token=1499607411-0xbbf3b9e10fd9d5186908fb83ae276f84028e2e05
Site:       Vimeo.com
Title:      Balloonfest от Topic on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       249.87 MiB (262009702 Bytes)

[DEBUG] get_content: https://vimeo.com/223642768
[DEBUG] get_content: http://player.vimeo.com/video/223642768
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4728/8/223642768/782804762.mp4?token=1499607424-0x668530a32327eefe19ee23fda656fe3e5f673123
Site:       Vimeo.com
Title:      Field of Vision - Conditioned Response from Field of Vision on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       185.65 MiB (194669298 Bytes)

[DEBUG] get_content: https://vimeo.com/223456771
[DEBUG] get_content: http://player.vimeo.com/video/223456771
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4691/8/223456771/781913187.mp4?token=1499607429-0x258fc2ae6b2c8dc200e91f1bcd461141c4df5fc7
Site:       Vimeo.com
Title:      Mouse from ELO films on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       283.55 MiB (297328310 Bytes)

[DEBUG] get_content: https://vimeo.com/223426078
[DEBUG] get_content: http://player.vimeo.com/video/223426078
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4685/8/223426078/781761766.mp4?token=1499607437-0xb74870eb747ce9c551b8fcf1e5ff6db2649ee1f9
Site:       Vimeo.com
Title:      What Youth Fairly Normal: Dane Reynolds from What Youth on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       503.33 MiB (527780227 Bytes)

[DEBUG] get_content: https://vimeo.com/223422820
[DEBUG] get_content: http://player.vimeo.com/video/223422820
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4684/8/223422820/781757588.mp4?token=1499607442-0xdacc6afe5e2904f065bb2b44ded953d8dc8cdd89
Site:       Vimeo.com
Title:      Vulfgang Rainstorme - Semblance (Official Music Video) from Yan Dan Wong on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       97.18 MiB (101903544 Bytes)

[DEBUG] get_content: https://vimeo.com/223270126
[DEBUG] get_content: http://player.vimeo.com/video/223270126
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4654/8/223270126/781020633.mp4?token=1499607449-0x62686f577b783507ae8fdadf44bf734ececa9753
Site:       Vimeo.com
Title:      H&ND from Pierre Emperador on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       82.55 MiB (86559406 Bytes)

[DEBUG] get_content: https://vimeo.com/223163255
[DEBUG] get_content: http://player.vimeo.com/video/223163255
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4632/8/223163255/780510624.mp4?token=1499607456-0xd8839c9291ba7a9118aa7675ce02f6485daa7ff9
Site:       Vimeo.com
Title:      Villa Antropoff from Kaspar Jancis on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       453.47 MiB (475495354 Bytes)

[DEBUG] get_content: https://vimeo.com/223149325
[DEBUG] get_content: http://player.vimeo.com/video/223149325
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4629/8/223149325/780441295.mp4?token=1499607466-0xabde2da428fb412a7c54de3b1d58227694c6de87
Site:       Vimeo.com
Title:      Deskercise from Foreign Fauna on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       19.15 MiB (20080423 Bytes)

[DEBUG] get_content: https://vimeo.com/223113394
[DEBUG] get_content: http://player.vimeo.com/video/223113394
you-get: 223113394 failed with HTTP Error 403: Forbidden
[DEBUG] get_content: https://vimeo.com/223080635
[DEBUG] get_content: http://player.vimeo.com/video/223080635
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4616/8/223080635/780108242.mp4?token=1499607479-0xdd0395a5a585cca7202088cd979ed90f84629945
Site:       Vimeo.com
Title:      Tokyo Gifathon from James Curran on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       36.3 MiB (38061223 Bytes)

[DEBUG] get_content: https://vimeo.com/223024285
[DEBUG] get_content: http://player.vimeo.com/video/223024285
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4604/8/223024285/779871095.mp4?token=1499607500-0x473f11436890873899ede6099366af31c5074527
Site:       Vimeo.com
Title:      Tough from Jennifer Zheng on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       133.77 MiB (140269309 Bytes)

[DEBUG] get_content: https://vimeo.com/222939711
[DEBUG] get_content: http://player.vimeo.com/video/222939711
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4587/8/222939711/779486326.mp4?token=1499607510-0xc85302305e77b40447a52429ed2e07a06685e815
Site:       Vimeo.com
Title:      They Will All Die in Space de Javier Chillon on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       359.79 MiB (377270499 Bytes)

[DEBUG] get_content: https://vimeo.com/222855070
[DEBUG] get_content: http://player.vimeo.com/video/222855070
you-get: 222855070 failed with HTTP Error 500: Internal Privoxy Error
[DEBUG] get_content: https://vimeo.com/222795354
[DEBUG] get_content: http://player.vimeo.com/video/222795354
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4559/8/222795354/778827612.mp4?token=1499607551-0x9824f5def864473457ae8ee094e3bff844b0070b
Site:       Vimeo.com
Title:      Morning coffee from GOBELINS pro on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       35.26 MiB (36969276 Bytes)

[DEBUG] get_content: https://vimeo.com/222729677
[DEBUG] get_content: http://player.vimeo.com/video/222729677
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4545/8/222729677/778517099.mp4?token=1499607558-0x44c5cccefce1abb0396ca89b7a857328d01db79d
Site:       Vimeo.com
Title:      FOOD von Michaela Mihalyiová on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       27.24 MiB (28568285 Bytes)

[DEBUG] get_content: https://vimeo.com/222718678
[DEBUG] get_content: http://player.vimeo.com/video/222718678
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4543/8/222718678/778463275.mp4?token=1499607562-0x2b01f09b6d81aeefaf486f8d6565406364a086f8
Site:       Vimeo.com
Title:      Their Names - Burning Star от Romain Laurent on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       78.9 MiB (82729371 Bytes)

[DEBUG] get_content: https://vimeo.com/222711836
[DEBUG] get_content: http://player.vimeo.com/video/222711836
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4542/8/222711836/778431154.mp4?token=1499607568-0x231b5f63051a31dd659367948e87260aeb98a64d
```

```
$ ./you-get -di 'https://vimeo.com/222855070'
[DEBUG] get_content: https://vimeo.com/222855070
[DEBUG] get_content: http://player.vimeo.com/video/222855070
[DEBUG] url_info: https://fpdl.vimeocdn.com/vimeo-prod-skyfire-std-us/01/4571/8/222855070/779088526.mp4?token=1499607946-0xb4c811576505bf91dc0f99486e266c4cc3fd983f
Site:       Vimeo.com
Title:      WEDNESDAY CAMPANELLA - BAKU MusicVideo from SOJIRO KAMATANI on Vimeo
Type:       MPEG-4 video (video/mp4)
Size:       207.82 MiB (217909840 Bytes)
```

fixed via ee506959511454f078586f3b10e96499e3eb9d2e
